pkg/api/customization/project/project_actions.go

```diff
@@ -0,0 +1,147 @@
+package project
+
+import (
+	"fmt"
+
+	"net/http"
+
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/parse"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	"github.com/rancher/types/client/management/v3"
+)
+
+func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+	resource.AddAction(apiContext, "setpodsecuritypolicytemplate")
+}
+
+type Handler struct {
+	ProjectClient v3.ProjectInterface
+}
+
+func (h *Handler) Actions(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	switch actionName {
+	case "setpodsecuritypolicytemplate":
+		return h.setPodSecurityPolicyTemplate(actionName, action, apiContext)
+	}
+
+	return errors.Errorf("bad action %v", actionName)
+}
+
+func (h *Handler) setPodSecurityPolicyTemplate(actionName string, action *types.Action, request *types.APIContext) error {
+	actionInput, err := parse.ReadBody(request.Request)
+	if err != nil {
+		return err
+	}
+
+	schema := request.Schemas.Schema(&managementschema.Version, client.PodSecurityPolicyTemplateProjectBindingType)
+	if schema == nil {
+		return fmt.Errorf("no %v store available", client.PodSecurityPolicyTemplateProjectBindingType)
+	}
+
+	binding, err := h.createOrUpdateBinding(request, schema, actionInput)
+	if err != nil {
+		return err
+	}
+
+	err = h.updateProjectPSPTID(request, actionInput)
+	if err != nil {
+		return err
+	}
+
+	request.WriteResponse(http.StatusOK, binding)
+
+	return nil
+}
+
+func (h *Handler) createOrUpdateBinding(request *types.APIContext, schema *types.Schema,
+	actionInput map[string]interface{}) (map[string]interface{}, error) {
+	bindings, err := schema.Store.List(request, request.Schema, &types.QueryOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var binding map[string]interface{}
+	for _, candidate := range bindings {
+		if candidate["projectId"] == request.ID {
```

Comment:
im not a stickler for using constants, but for api fields, we genterate them and they should be used as a safegaurd against the field changing
```
client.PodSecurityPolicyTemplateProjectBindingFieldProjectID
```